cogeo_mosaic/backends/sqlite.py

```diff
@@ -0,0 +1,237 @@
+                (self._metadata_quadkey,),
+            ).fetchall()
+
+        return [r["quadkey"] for r in rows]
+
+    def write(self, overwrite: bool = False):
+        """Write mosaicjson document to an SQLite database.
+
+        Args:
+            overwrite (bool): delete old mosaic items inthe Table.
+
+        Returns:
+            dict: dictionary with metadata constructed from the sceneid.
+
+        Raises:
+            MosaicExistsError: If mosaic already exists in the Table.
+
+        """
+        if self._mosaic_exists():
+            if not overwrite:
+                raise MosaicExistsError(
+                    f"'{self.mosaic_name}' Table already exists in {self.db_name}, use `overwrite=True`."
+                )
+            self.delete()
+
+        logger.debug(f"Creating '{self.mosaic_name}' Table in {self.db_name}.")
+        with self.db:
+            self.db.execute(
+                f"CREATE TABLE {self.mosaic_name} (quadkey TEXT NOT NULL, value JSON NOT NULL);",
+            )
+
+        logger.debug(f"Adding items in '{self.mosaic_name}' Table.")
+        items = []
+        items.append((self._metadata_quadkey, self.metadata.dict()))
+        for qk, assets in self.mosaic_def.tiles.items():
+            items.append((qk, {"assets": assets}))
+
+        with self.db:
+            self.db.executemany(
+                f"INSERT INTO {self.mosaic_name} (quadkey, value) VALUES (?, ?);", items
+            )
+
+        return
+
+    def _update_quadkey(self, quadkey: str, dataset: List[str]):
+        """Update single quadkey in Table."""
+        with self.db:
+            self.db.execute(
+                f"UPDATE {self.mosaic_name} SET value = ? WHERE quadkey=?;",
+                ({"assets": dataset}, quadkey),
+            )
+
+    def _update_metadata(self):
+        """Update bounds and center."""
+        meta = self.metadata.dict()
+        meta["mosaicId"] = self.mosaic_name
+        with self.db:
+            self.db.execute(
+                f"UPDATE {self.mosaic_name} SET value = ? WHERE quadkey=?;",
+                (meta, self._metadata_quadkey),
+            )
+
+    def update(
+        self,
+        features: Sequence[Dict],
+        add_first: bool = True,
+        quiet: bool = False,
+        **kwargs,
+    ):
+        """Update existing MosaicJSON on backend."""
+        logger.debug(f"Updating {self.mosaic_name}...")
+
+        new_mosaic = self.mosaic_def.from_features(
+            features,
+            self.mosaic_def.minzoom,
+            self.mosaic_def.maxzoom,
+            quadkey_zoom=self.quadkey_zoom,
+            quiet=quiet,
+            **kwargs,
+        )
+
+        for quadkey, new_assets in new_mosaic.tiles.items():
+            tile = mercantile.quadkey_to_tile(quadkey)
+            assets = self.assets_for_tile(*tile)
+            assets = [*new_assets, *assets] if add_first else [*assets, *new_assets]
+            self._update_quadkey(quadkey, assets)
```

Comment:
Here we could maybe use `executemany` and also maybe to SQL operation directly 🤷‍♂️